src/aura/BGE_BatchGiftEntryTab/BGE_BatchGiftEntryTabHelper.js

```diff
@@ -87,16 +100,17 @@
 
     /**
      * @description: checks that user has all necessary permissions and then launches modal or displays error
+     * @param: sourceBatchId Id of Batch record to be copied - optional
      */
-    checkFieldPermissions: function(component) {
+    checkFieldPermissions: function(component, sourceBatchId) {
```

Comment:
This is a suggested refactor for something that predates this work item, but to address the confusion of opening the wizard model by calling a function called "checkFieldPermissions", I wonder if we can pass in the openNewBatchWizard() as a callback to checkFieldPermissions - 

Called with: 
    checkFieldPermissions(component, this.openNewBatchWizard(component, sourceBatchId));

And the function def here: 
    checkFieldPermissions: function(component, onSuccess) {

Honestly not sure if this is worth it just to make it a bit clearer how hitting that action opens the Wizard after permissions check passes, but wanted to bring it up

Reply:
We haven't used that as a pattern before, but I'd be interested to see if this works and still prevents a double click! If it doesn't, we could name things more explicitly "checkFieldPermissionsToOpenWizard" ?